src/components/auth/Register.js

```diff
@@ -0,0 +1,169 @@
+import React from "react";
+import { axiosWithAuth } from "../../utils/axiosWithAuth.js";
+
+// styling imports
+import "./_Register.scss";
+import GroaWhite from "./GroaWhite.png";
+
+class Register extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      user: {
+        email: "",
+        user_name: "",
+        password: "",
+        confirmpassword: ""
+      },
+      submitted: false,
+      errorStatus: false
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    const { user } = this.state;
+    this.setState({
+      user: {
+        ...user,
+        [name]: value
+      }
+    });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    let user = {
+      user_name: this.state.user.user_name,
+      password: this.state.user.password,
+      email: this.state.user.email
+    };
+    this.setState({ submitted: true });
+    if (
+      this.state.user.email &&
+      this.state.user.user_name &&
+      this.state.user.password &&
+      this.state.user.confirmpassword
+    ) {
+      axiosWithAuth()
+        .post("https://api.groa.us/api/users/register", user)
+        .then(() => {
+          user = {
+            user_name: this.state.user.user_name,
+            password: this.state.user.password
+          };
+          axiosWithAuth()
+            .post("https://api.groa.us/api/users/login", user)
+            .then(res2 => {
+              const userid = res2.data.id;
+              localStorage.setItem("token", res2.data.token);
+              this.props.history.push("/dashboard", { userid: userid });
+            })
+            .catch(err2 => {
+              console.log(err2);
+            });
+        })
+        .catch(err => {
+          console.log("Registration Error", err);
+          this.setState({ errorStatus: true });
+        });
+    }
+  }
+
+  render() {
+    const { user, submitted } = this.state;
+    return (
+      <div className="container">
+        <div className="boxLeft">
+          <img className="logo" src={GroaWhite} alt="groa logo" />
+        </div>
+        <div className="boxRight">
+          <form className="form" onSubmit={this.handleSubmit}>
```

Comment:
Just a reminder for me, we'll have to figure out a plan for error handling soon.